src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts

```diff
@@ -273,10 +272,12 @@ export class EventDateTimePickerComponent implements OnInit, OnChanges {
     }
   }
 
-  private checkEndTime(time: string): void {
-    if (time) {
+  private checkEndTime(time: string, curTime?: Array<string>): void {
+    if (time && curTime) {
+      const checkTime = time.split(':');
+      this.timeArrStart = [...this.timeArr.slice(+curTime[0] + 1, +checkTime[0])];
+    } else if (time) {
```

Comment:
if (!time) return;
const checkTime = time.split(':');
if (curTime) {
this.timeArrStart = [...this.timeArr.slice(+curTime[0] + 1, +checkTime[0])];
} else {
this.timeArrStart = [...this.timeArr.slice(0, +checkTime[0])];
}